.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,59 @@
+name: Static Analysis
+
+on:
+  schedule:
+    - cron: "0 2 * * 1"
+
+  push:
+    branches: [main]
+    paths:
+      - "**.go"
+      - "go.mod"
+      - "go.sum"
+      - ".golangci.yaml"
```

Comment:
Please remove this section, which causes triggering at the wrong times. Remember, we decided to ONLY trigger this workflow weekly or when requested. See https://docs.github.com/en/actions/reference/workflows-and-actions/events-that-trigger-workflows#push for the meaning of triggering on "push".